tests/neo4j/txrun.py

```diff
@@ -40,3 +38,100 @@ def test_does_not_update_last_bookmark_on_rollback(self):
+        tx = session.beginTransaction()
+        with self.assertRaises(types.responses.DriverError):
+            tx.run("RETURN").next()
+
+    def test_should_not_rollback_a_rollbacked_tx(self):
+        if get_driver_name() in ["go"]:
+            self.skipTest('Does not raise the exception')
+        self._session = self._driver.session("w")
+        tx = self._session.beginTransaction()
+        tx.run('CREATE (:TXNode1)').consume()
+        tx.rollback()
+        with self.assertRaises(types.responses.DriverError):
+            tx.rollback()
+
+    def test_should_not_rollback_a_commited_tx(self):
+        if get_driver_name() in ["go"]:
+            self.skipTest('Does not raise the exception')
+        session = self._driver.session(accessMode="w")
+        tx = session.beginTransaction()
+        tx.run('CREATE (:TXNode1)').consume()
+        tx.commit()
+        with self.assertRaises(types.responses.DriverError):
+            tx.rollback()
+
+    def test_should_not_commit_a_commited_tx(self):
+        if get_driver_name() in ["go"]:
+            self.skipTest('Does not raise exception')
+        self._session = self._driver.session(accessMode="w")
+        tx = self._session.beginTransaction()
+        tx.run('CREATE (:TXNode1)').consume()
+        tx.commit()
+        with self.assertRaises(types.responses.DriverError):
+            tx.commit()
+
+    def test_should_not_run_valid_query_in_invalid_tx(self):
+        if get_driver_name() in ["python"]:
+            self.skipTest("executes the second RUN")
+        if get_driver_name() in ["go"]:
+            self.skipTest('Could not rollback transaction')
+
+        session = self._driver.session(accessMode="w")
+        tx = session.beginTransaction()
+        with self.assertRaises(types.responses.DriverError):
+            tx.run("NOT CYPHER").consume()
+
+        with self.assertRaises(types.responses.DriverError):
+            tx.run("RETURN 42").next()
+
+        tx.rollback()
+
+    def test_should_fail_run_in_a_commited_tx(self):
+        self._session = self._driver.session(accessMode="w")
+        tx = self._session.beginTransaction()
+        tx.commit()
+        with self.assertRaises(types.responses.DriverError):
+            tx.run("RETURN 42").next()
+
+    def test_should_fail_run_in_a_rollbacked_tx(self):
+        self._session = self._driver.session(accessMode="w")
+        tx = self._session.beginTransaction()
+        tx.rollback()
+        with self.assertRaises(types.responses.DriverError):
+            tx.run("RETURN 42").next()
+
+    def test_should_fail_to_run_query_for_unreacheable_bookmark(self):
+        if get_driver_name() in ["go", "python"]:
+            self.skipTest("Does not throws exception on run unreachable bm")
+        session = self._driver.session(accessMode="w")
+        tx1 = session.beginTransaction()
+        result = tx1.run('CREATE ()')
+        result.consume()
+        tx1.commit()
+        lastBookmark = session.lastBookmarks()[0]
```

Comment:
```suggestion
        tx1.commit()
        lastBookmarks = session.lastBookmarks()
        assert len(lastBookmarks) == 1
        lastBookmark = lastBookmarks[0]
```
Makes the test fail instead of error if the driver doesn't return a bookmark (or multiple)